atlp-rwanda/stackup2-barefoot-backend
# #171380741 Setup API documentation

#### What does this PR do?
Setup API documentation using Swagger
#### Description of Task to be completed?
* Initialize an express server in index.js file
* Create a basic swagger.json spec file
* Install `express`, `swagger-ui-express` and `babel` libraries
* Create a sample route at `/api/v1`
* Create `.babelrc` file
#### How should this be manually tested?
1. Open your terminal and run `git clone https://github.com/Stackup-Rwanda/stackup2-barefoot-backend/`
2. run `cd stackup2-barefoot-backend` to go into the root directory of the project
3. run `git checkout -b ch-setup-api-documentation-171380741 origin/ch-setup-api-documentation-171380741` to pull this branch and switch to it
4. run `npm install` to install the dependencies
5. run `npm run dev` to start the server
6. Open your browser and navigate to this link: [localhost:3000/public/api-docs](http://localhost:3000/public/api-docs)
7. Click on the highlighted `GET` route, then click on the `Try it out` button then `execute` button and you should see a server response of status `200` and a message of `Welcome!` 
#### Any background context you want to provide?
* You need to install [Node.js](https://nodejs.org/en/) and [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) before testing.
#### What are the relevant pivotal tracker stories?
[#171380741](https://www.pivotaltracker.com/story/show/171380741)
#### Screenshots (if appropriate)
N/A
#### Questions:
